tests/lms/views/test_lti_launches.py

```diff
@@ -33,3 +36,15 @@ def test_render_unauthorized_for_students(self, lti_launch_request, module_item_
         lti_launch_request.params['roles'] = 'urn:lti:role:ims/lis/Learner'
         value = lti_launches(lti_launch_request)
         assert 'This page has not yet been configured' in value.body.decode()
+
+    def test_raises_for_missing_context_id_param(self, lti_launch_request):
+        lti_launch_request.params.pop('context_id', None)
+
+        with pytest.raises(MissingLTILaunchParamError, match="LTI data parameter context_id is required for launch."):
+            lti_launches(lti_launch_request)
+
+    def test_raises_for_missing_resource_link_id_param(self, lti_launch_request):
+        lti_launch_request.params.pop('resource_link_id', None)
```

Comment:
I think these two `pop()`s can be replaced with `del lti_launch_request.params["resource_link_id"]`, `pop()` actually returns the value from the dict (or `None`) but we don't need that here